### PR TITLE
[Gemma4] Fix the bug of `use_qk_norm` accidently set to False in Gemma 4 configs 

### PR DESCRIPTION
**Pull request by Dr-Left**
# Description

This change resolves an issue with Gemma 4 attention weight initialization and QK normalization:

1. In `src/maxtext/layers/attentions.py`, `init_query_w` now checks `self.use_qk_norm` in addition to `self.config.use_qk_norm`. When `use_qk_norm` is enabled (or `query_pre_attn_scalar != 1.0`), `depth_scaling` is set to 1.0 instead of `sqrt(head_dim)`. This prevents dividing the initial query projection weights by `sqrt(head_dim)`, ensuring unscaled canonical weight initialization for Gemma 4.
2. In Gemma 4 model configs (`gemma4-26b.yml`, `gemma4-31b.yml`, `gemma4-e2b.yml`, `gemma4-e4b.yml`), explicitly enabled `use_qk_norm: true`.
3. In `gemma4-e2b.yml` and `gemma4-e4b.yml`, set `scan_layers: false` since per-layer embeddings and variable layer configs are not compatible with layer scanning.
4. Added `tests/unit/gemma4_canonical_test.py` to verify that canonical attention scaling (`query_pre_attn_scalar = 1.0`), `use_qk_norm`, and unscaled weight initialization hold for both dense/MoE Gemma 4 and small Gemma 4 models.

# Tests
Smoke test on V5p-256 x 2 slices (Data parallel training of Gemma4-26B-A4B)
Before fix: http://tb/281018541  
After fix: http://tb/281083608   [log_link](https://pantheon.corp.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22cloud-tpu-multipod-dev%22%0Aresource.labels.location%3D%22europe-west4%22%0Aresource.labels.cluster_name%3D%22mlperf-v5p%22%0Aresource.labels.namespace_name%3D%22default%22%0Aresource.labels.pod_name:%22dp-gm4-fix-slice-job-0-0-%22%0Aseverity%3E%3DDEFAULT;storageScope=project;cursorTimestamp=2026-08-18T22:43:14.953903353Z;duration=P1D?project=cloud-tpu-multipod-dev&e=13803378)

Added unit test:
- `tests/unit/gemma4_canonical_test.py`

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/layers/attentions.py [hunk 1]
```diff
@@ -665,8 +665,10 @@ def init_query_w(self, inputs_q_shape: Tuple) -> nnx.Module:
     #       linear transformations, which is equivalent under Adafactor.
     # We disable depth_scaling when using qk_norm or a query_pre_attn_scalar
     # to avoid applying scaling twice.
-    if getattr(self.config, "use_qk_norm", False) or (
-        self.query_pre_attn_scalar is not None and self.query_pre_attn_scalar != 1.0
+    if (
+        self.use_qk_norm
+        or getattr(self.config, "use_qk_norm", False)
+        or (self.query_pre_attn_scalar is not None and self.query_pre_attn_scalar != 1.0)
     ):
```
[the remark this answers is not in the document]
Comment (Dr-Left) [reply src/maxtext/layers/attentions.py]: Fixed in new push. default to None for all attentions and override by config.use_qk_norm

File: tests/unit/gemma4_canonical_test.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+          config=config,
+          mesh=mesh,
+          model_mode=common_types.MODEL_MODE_PREFILL,
+          layer_idx=0,
+          rngs=rngs,
+      )
+      self.assertEqual(layer.self_attention.query_pre_attn_scalar, 1.0)
+      self.assertTrue(layer.self_attention.use_qk_norm)
+
+      q_kernel = layer.self_attention.query.kernel[...]
```
[the remark this answers is not in the document]
Comment (Dr-Left) [reply tests/unit/gemma4_canonical_test.py:85]: Replacing kernel[...] with kernel.value would re-introduce DeprecationWarnings across the codebase.

File: tests/unit/gemma4_canonical_test.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        attention_type=gemma4.AttentionType.LOCAL_SLIDING,
+        layer_idx=0,
+    )
+    # Canonical Gemma 4 attention uses query_pre_attn_scalar = 1.0 (unscaled logits)
+    self.assertEqual(layer.self_attention.query_pre_attn_scalar, 1.0)
+    self.assertTrue(layer.self_attention.use_qk_norm)
+
+    # Initial query weights should NOT be divided by sqrt(head_dim) = 16.0
+    # Expected standard deviation for fan_in=2816 is 1/sqrt(2816) ~= 0.0188
+    q_kernel = layer.self_attention.query.kernel[...]
```
[the remark this answers is not in the document]
Comment (Dr-Left) [reply tests/unit/gemma4_canonical_test.py:60]: Replacing kernel[...] with kernel.value would re-introduce DeprecationWarnings across the codebase.
